### PR TITLE
Fix for #39  Nullpointer exception on searching for non-existing term

### PR DESCRIPTION
**Pull request by Xyclade**
Included 2 unit tests to cover the bugfix

### CODE REVIEW

File: SmileNLP/src/main/java/smile/nlp/SimpleCorpus.java [hunk 1]
```diff
@@ -234,23 +234,37 @@ public int getBigramFrequency(Bigram bigram) {
 
     @Override
     public Iterator<Text> search(String term) {
-    	ArrayList<Text> hits = new ArrayList<Text>(invertedFile.get(term));
-        return hits.iterator();
+        if (invertedFile.containsKey(term))
```
Comment (haifengl) [inline SmileNLP/src/main/java/smile/nlp/SimpleCorpus.java]: Our code format is 

if () {
}

I will merge the request and revise it by myself. Please use the same format next time. Thank you very much!

**Comment by Xyclade** [comment]
My bad, should've formatted it before committing, I'll do so next time :+1:

**Comment by haifengl** [comment]
No problem at all. Thank you very much for contribution! Really appreciate!

**Comment by ravikuril** [comment]
What does it do I can't understand can u please give me brief introduction
and how to use this module ......help needed

**Comment by haifengl** [comment]
@ravikuril our wiki has some tutorials. here is one:

https://github.com/haifengl/smile/wiki/Tutorial:-A-Gentle-Introduction-to-SmileMiner

**Comment by Xyclade** [comment]
@ravikuril  additional to the tutorials on the wiki, the library can be used for processing natural languages to a more structured set of data, by stemming, removing stopwords, finding bigrams etc. In machine learning you tend to use NLP in combination with another form of learning, to make something powerful. 

I suggest you look up some of the general notions of NLP (be careful that you search for Natural Language processing, and not for Neuro-linguistic programming) , and look at the examples from the tutorials. If you understand the examples, then getting the NLP part to work should be easy as it follows the same principles :+1:

**Pull request merged**